graphbolt/include/graphbolt/csc_sampling_graph.h

```diff
@@ -161,25 +65,14 @@ class CSCSamplingGraph : public torch::CustomClassHolder {
   /** @brief Get the index tensor. */
   const torch::Tensor Indices() const { return indices_; }
 
-  /** @brief Check if the graph is heterogeneous. */
-  inline bool IsHeterogeneous() const { return hetero_info_ != nullptr; }
-
   /** @brief Get the node type offset tensor for a heterogeneous graph. */
-  inline const torch::Tensor NodeTypeOffset() const {
-    return hetero_info_->node_type_offset;
-  }
-
-  /** @brief Get the list of node types for a heterogeneous graph. */
-  inline NodeTypeList& NodeTypes() const { return hetero_info_->node_types; }
-
-  /** @brief Get the list of edge types for a heterogeneous graph. */
-  inline const EdgeTypeList& EdgeTypes() const {
-    return hetero_info_->edge_types;
+  inline const torch::optional<torch::Tensor> NodeTypeOffset() const {
+    return node_type_offset_;
```

Comment:
double check which is the less error-prone option when finding node type id from a node id.

- include both 0, num_nodes
- include 0, exclude num_nodes
- exclude 0, include num_nodes
- exclude both

Reply:
include both 0, num_nodes is more convenient, because:
1. It's more compatiable with c++ `upper_bound` function
2. It's easy to check out of bound nodes.